### DISCUSSION TITLE
Terraform deployment - Cloud-init issues

### DISCUSSION THREAD

**Discussion by velakuruday** [telekom-security/tpotce#1211] [Q&A] [answered]
Hey there,
I tried to deploy tpot using the provided terraform script. I'm getting the following issues (cloud init logs): 

 ___           _        _ _ _                     _             
|_ _|_ __  ___| |_ __ _| | (_)_ __   __ _   _ __ | | ____ _ ___ 
 | || '_ \/ __| __/ _` | | | | '_ \ / _` | | '_ \| |/ / _` / __|
 | || | | \__ \ || (_| | | | | | | | (_| | | |_) |   < (_| \__ \
|___|_| |_|___/\__\__,_|_|_|_|_| |_|\__, | | .__/|_|\_\__, |___/
                                    |___/  |_|        |___/     
/root/tpot/iso/installer/install.sh: line 698: npm: command not found
/root/tpot/iso/installer/install.sh: line 699: pip3: command not found
  ____ _             _               _____     ____       _   
 / ___| | ___  _ __ (_)_ __   __ _  |_   _|   |  _ \ ___ | |_ 
| |   | |/ _ \| '_ \| | '_ \ / _` |   | |_____| |_) / _ \| __|
| |___| | (_) | | | | | | | | (_| |   | |_____|  __/ (_) | |_ 
 \____|_|\___/|_| |_|_|_| |_|\__, |   |_|     |_|   \___/ \__|
                             |___/                            
Cloning into '/opt/tpot'...
  ____                _                                         
 / ___|_ __ ___  __ _| |_ ___    __ _ _ __ ___  _   _ _ __  ___ 
| |   | '__/ _ \/ _` | __/ _ \  / _` | '__/ _ \| | | | '_ \/ __|
| |___| | |  __/ (_| | ||  __/ | (_| | | | (_) | |_| | |_) \__ \
 \____|_|  \___|\__,_|\__\___|  \__, |_|  \___/ \__,_| .__/|___/
                                |___/                |_|        
Adding group `tpot' (GID 2000) ...
Done.
Adding group `tpotlogs' (GID 1001) ...
Done.
  ____                _                             
 / ___|_ __ ___  __ _| |_ ___   _   _ ___  ___ _ __ 
| |   | '__/ _ \/ _` | __/ _ \ | | | / __|/ _ \ '__|
| |___| | |  __/ (_| | ||  __/ | |_| \__ \  __/ |   
 \____|_|  \___|\__,_|\__\___|  \__,_|___/\___|_|   
                                                    
Adding system user `tpot' (UID 2000) ...
Adding new user `tpot' (UID 2000) with group `tpot' ...
Not creating home directory `/home/tpot'.
 ____       _     _               _                              
/ ___|  ___| |_  | |__   ___  ___| |_ _ __   __ _ _ __ ___   ___ 
\___ \ / _ \ __| | '_ \ / _ \/ __| __| '_ \ / _` | '_ ` _ \ / _ \
 ___) |  __/ |_  | | | | (_) \__ \ |_| | | | (_| | | | | | |  __/
|____/ \___|\__| |_| |_|\___/|___/\__|_| |_|\__,_|_| |_| |_|\___|
                                                                 
    _       _  _           _                      _       
   / \   __| |(_)_   _ ___| |_   _ __   ___  _ __| |_ ___ 
  / _ \ / _` || | | | / __| __| | '_ \ / _ \| '__| __/ __|
 / ___ \ (_| || | |_| \__ \ |_  | |_) | (_) | |  | |_\__ \
/_/   \_\__,_|/ |\__,_|___/\__| | .__/ \___/|_|   \__|___/
            |__/                |_|                       
[Socket]
ListenStream=
ListenStream=64294

Port 64295
Match Group tpotlogs
        PermitOpen 127.0.0.1:64305
        ForceCommand /usr/bin/false

sed: can't read /etc/pam.d/cockpit: No such file or directory
 ____ _____  _    _   _ ____    _    ____  ____  
/ ___|_   _|/ \  | \ | |  _ \  / \  |  _ \|  _ \ 
\___ \ | | / _ \ |  \| | | | |/ _ \ | |_) | | | |
 ___) || |/ ___ \| |\  | |_| / ___ \|  _ <| |_| |
|____/ |_/_/   \_\_| \_|____/_/   \_\_| \_\____/ 
                                                 
 ____        _ _   _                                 
|  _ \ _   _| | | (_)_ __ ___   __ _  __ _  ___  ___ 
| |_) | | | | | | | | '_ ` _ \ / _` |/ _` |/ _ \/ __|
|  __/| |_| | | | | | | | | | | (_| | (_| |  __/\__ \
|_|    \__,_|_|_| |_|_| |_| |_|\__,_|\__, |\___||___/
                                     |___/           
/root/tpot/iso/installer/install.sh: line 780: docker: command not found
/root/tpot/iso/installer/install.sh: line 780: docker: command not found
/root/tpot/iso/installer/install.sh: line 780: docker: command not found
/root/tpot/iso/installer/install.sh: line 780: docker: command not found
/root/tpot/iso/installer/install.sh: line 780: docker: command not found
/root/tpot/iso/installer/install.sh: line 780: docker: command not found
/root/tpot/iso/installer/install.sh: line 780: docker: command not found

Seems like the packages (npm, docker etc.) were not automatically installed when using the cloud init. 
Any ideas on what might be going on?

**Comment by t3chn0m4g3** [comment]
@velakuruday Terraform is a community developed feature.
@tmariuss @shaderecker Can you help?

**Comment by shaderecker** [comment, marked as answer]
Hi @velakuruday 
not really a cloud-init issue as I  see it.
Those packages get installed by the tpot installer script, not by cloud-init.
Install happens [here](https://github.com/telekom-security/tpotce/blob/master/iso/installer/install.sh#L309), package list is determined [here](https://github.com/telekom-security/tpotce/blob/master/iso/installer/install.sh#L25-L30) and the package list can be found [here](https://github.com/telekom-security/tpotce/blob/master/packages.txt).
So maybe there is an issue during the package installation itself or an issue with the used package mirror.
Could you check the `/install.log` & `/install.err` for any hints?
